types/share_merging.go

```diff
@@ -212,71 +212,65 @@ func parseMsgShares(shares [][]byte) ([]Message, error) {
 	if len(shares) == 0 {
 		return nil, nil
 	}
-
-	// set the first nid and current share
-	nid := shares[0][:consts.NamespaceSize]
-	currentShare := shares[0][consts.NamespaceSize:]
-	// find and remove the msg len delimiter
-	currentShare, msgLen, err := ParseDelimiter(currentShare)
-	if err != nil {
-		return nil, err
+	// msgs returned
```

Comment:
If I understand right, this function wasn't able to parse padding, now it is, right?
So, creating a block using the non-interactive defaults rules is already implemented somewhere?

Reply:
> So, creating a block using the non-interactive defaults rules is already implemented somewhere?

While I have some very crude local experimental branches with pieces here and there that are not ready for the light of day, mostly yes. I did however feel comfortable enough to spin this out, especially since the most needed thing is just an easy way to add padding.

https://github.com/celestiaorg/celestia-specs/blob/master/src/rationale/message_block_layout.md#caveats

Reply:
I'm also okay with not merging this until the app implementation is ready

Reply:
It's good. It's better to merge PRs and not stockpile them. I guess merging is better :rocket: